bcipy/signal/process/transforms.py

```diff
@@ -0,0 +1,74 @@
+from scipy.signal import butter, filtfilt, iirnotch, sosfilt
+import numpy as np
+from typing import Optional
+
+
+class Composition:
+    def __init__(self, *transforms):
+        self.transforms = transforms
+
+    def __call__(self, data: np.ndarray, fs: Optional[int] = None):
+        for t in self.transforms:
+            data, fs = t(data, fs)
+        if fs is None:
+            return data
+        return data, fs
+
+
+class Notch:
+    def __init__(self, sample_rate_hz, remove_freq_hz=60.0, quality_factor=30):
+        remove_freq_hz = remove_freq_hz / (sample_rate_hz / 2)
+        self.b, self.a = iirnotch(remove_freq_hz, quality_factor)
+
+    def __call__(self, data: np.ndarray, fs: Optional[int] = None):
+        if fs is None:
+            return filtfilt(self.b, self.a, data)
+        return filtfilt(self.b, self.a, data), fs
+
+
+class Bandpass:
+    def __init__(self, lo, hi, sample_rate_hz, order=5):
+        nyq = 0.5 * sample_rate_hz
+        lo, hi = lo / nyq, hi / nyq
+        self.sos = butter(order, [lo, hi], analog=False, btype="band", output="sos")
+
+    def __call__(self, data: np.ndarray, fs: Optional[int] = None):
+        if fs is None:
+            return sosfilt(self.sos, data)
+        return sosfilt(self.sos, data), fs
+
+
+class Downsample:
+    def __init__(self, factor: int = 2):
+        self.factor = factor
+
+    def __call__(self, data: np.ndarray, fs: Optional[int] = None):
+        if fs is None:
+            return data[:, :: self.factor]
+        return data[:, :: self.factor], fs // self.factor
+
+
+class Lambda:
```

Comment:
Is there a more specific label we can give f? 

In this case, should f return fs? What if it transformed it in some way that impacts it?

Reply:
Hmm good point - maybe just leave out this transform for now, and add it back when we have a use case in mind?

(In general though probably `f` should always operate on both `data` and `fs`, so that the user can keep these two items in sync)